libs/blocks/text/text.css

```diff
@@ -100,7 +100,7 @@
   position: relative;
 }
 
-.text-block .icon-list-item .icon.margin-right:not(.margin-left) { /* target first node only */
+.text-block .icon-list-item .icon.node-index-first:not(.node-index-last) { /* target first node only */
```

Comment:
Yes, I am trying to support previous use case where we used margin-right, margin-left. etc and now this is handled a bit different, I don't think the not() part is necessary since there will never be both. I'm going to update and test.

Reply:
FYI, I've simplified this and not using :not